### PR TITLE
fix(admin): admin open contribution edit button

### LINKED ISSUE #2803: 🐛 [Bug][Admin] open contribution missing edit button
## 🐛 Bugreport

Open contribution misses the edit button

### PR DESCRIPTION
**Pull request by Elweyn**
### Issues

- fixes #2803 

### Todo

- [X] None

### CODE REVIEW

**Review by ogerly — DISMISSED** [review]

File: backend/src/graphql/resolver/ContributionResolver.ts [hunk 1]
```diff
@@ -314,41 +314,28 @@ export class ContributionResolver {
...
-    }
-    if (emailContact.deletedAt || emailContact.user.deletedAt) {
-      throw new LogError('User was deleted', email)
-    }
 
     const moderator = getUser(context)
 
     const contributionToUpdate = await DbContribution.findOne({
       where: { id, confirmedAt: IsNull(), deniedAt: IsNull() },
+      relations: ['user'],
```
Comment (Mogge) [inline backend/src/graphql/resolver/ContributionResolver.ts:326]: @ulfgebhardt I need the user relation only for the event here. Why did you insist on a user object?

Comment (Mogge) [reply backend/src/graphql/resolver/ContributionResolver.ts:326]: Furthermore. This is a contradiction to the graphql idea. I should only query the user, if I really need it. No I am forced by the event to query the user always

Comment (ulfgebhardt) [reply backend/src/graphql/resolver/ContributionResolver.ts:326]: `{ id: myuserId } as DBUser`

File: admin/src/components/EditCreationFormular.vue [hunk 2]
```diff
@@ -96,41 +96,31 @@ export default {
...
       rangeMin: 0,
-      rangeMax: 1000,
-      selected: '',
+      selected: this.selectedComputed,
+      userId: this.item.userId,
     }
   },
   methods: {
     submitCreation() {
+      // console.log('submitCreation', this.selected)
```
Comment (ulfgebhardt) [inline admin/src/components/EditCreationFormular.vue:111]: remove

File: backend/src/graphql/resolver/ContributionResolver.test.ts [hunk 3]
```diff
@@ -2080,15 +2077,14 @@ describe('ContributionResolver', () => {
           // stephen@hawking.uk: [1000, 1000, 1000] - deleted
           // garrick@ollivander.com: [1000, 1000, 1000] - not activated
 
-          describe('user for creation to update does not exist', () => {
+          describe.skip('user for creation to update does not exist', () => {
```
Comment (ulfgebhardt) [inline backend/src/graphql/resolver/ContributionResolver.test.ts:2080]: why skip? If the user status is no criteria anymore lets remove the test?
Is there implications for the operation if the user was deleted/does not exist?

File: backend/src/graphql/resolver/ContributionResolver.test.ts [hunk 4]
```diff
@@ -2106,15 +2102,14 @@ describe('ContributionResolver', () => {
             })
           })
 
-          describe('user for creation to update is deleted', () => {
+          describe.skip('user for creation to update is deleted', () => {
```
Comment (ulfgebhardt) [inline backend/src/graphql/resolver/ContributionResolver.test.ts:2105]: why skip?

File: backend/src/graphql/resolver/ContributionResolver.test.ts [hunk 5]
```diff
@@ -2158,15 +2152,14 @@ describe('ContributionResolver', () => {
             })
           })
 
-          describe('user email does not match creation user', () => {
+          describe.skip('user email does not match creation user', () => {
```
Comment (ulfgebhardt) [inline backend/src/graphql/resolver/ContributionResolver.test.ts:2155]: why skip?

File: backend/src/graphql/resolver/ContributionResolver.ts (hunk 1 above)
Comment (ulfgebhardt) [inline backend/src/graphql/resolver/ContributionResolver.ts:326]:
```suggestion
```

File: backend/src/graphql/resolver/ContributionResolver.ts [hunk 6]
```diff
@@ -372,10 +359,8 @@ export class ContributionResolver {
     result.memo = contributionToUpdate.memo
     result.date = contributionToUpdate.contributionDate
 
-    result.creation = await getUserCreation(emailContact.user.id, clientTimezoneOffset)
-
     await EVENT_ADMIN_CONTRIBUTION_UPDATE(
-      emailContact.user,
+      contributionToUpdate.user,
```
Comment (ulfgebhardt) [inline backend/src/graphql/resolver/ContributionResolver.ts:363]:
```suggestion
      { id: contributionToUpdate.userId } as DbUser,
```

File: backend/src/graphql/resolver/ContributionResolver.test.ts (hunk 4 above)
Comment (Mogge) [reply backend/src/graphql/resolver/ContributionResolver.test.ts:2105]: Same. It does not matter if the user exists or is deleted, as I do not query the user anymore

File: backend/src/graphql/resolver/ContributionResolver.test.ts (hunk 3 above)
Comment (ulfgebhardt) [inline backend/src/graphql/resolver/ContributionResolver.test.ts:2080]: I would define that you can update an contribution even tho an user does not exist or is deleted. Therefore I would remove the two tests completely.

File: admin/src/components/CreationFormular.vue [hunk 7]
```diff
@@ -129,21 +125,21 @@ export default {
...
+      userId: this.item.userId,
     }
   },
   methods: {
     updateRadioSelected(name) {
       // do we want to reset the memo everytime the month changes?
       this.text = this.$t('creation_form.creation_for') + ' ' + name.short + ' ' + name.year
       this.rangeMin = 0
-      this.rangeMax = name.creation
+      this.rangeMax = Number(name.creation)
```
Comment (ulfgebhardt) [inline admin/src/components/CreationFormular.vue:136]: I stumbled over this function - main reason was the parameter `name` which is not correct in my opinion - I think it should be `selected` or `option`. It is a bit out of scope, but this refactoring would increase readability alot I think. If you don't want to do this, please just close.

**Review by ulfgebhardt — CHANGES_REQUESTED** [review]
image
image
image

I cannot test this, since I can only edit Contributions in the Admin interface I created there. But apparently there is a side effect of your code that affects `adminCreateContribution` 
image

So besides minor code rather questions then suggestions this must be fix or explained why this behaviour is like that. Furthermore the tests should reflect this problem with a failure.

**Review by Mogge — COMMENTED** [review]
I have no idea where all the eslint changes are coming from ...

**Review by ulfgebhardt — APPROVED** [review]
❤️

**Pull request merged**